src/container/srv_target.c

```diff
@@ -1209,10 +1221,13 @@ cont_child_destroy_one(void *vin)
 		}
 		ABT_mutex_unlock(cont->sc_mutex);
 
-		/* Make sure rebuild has stopped */
+		/**
+		 * After @sc_stopping is 1, all cont lookup will fail, so we wait
+		 * existed ds_cont users exit, it might come from rebuild, dtx.
+		 */
 		ABT_mutex_lock(cont->sc_mutex);
-		if (cont->sc_rebuilding)
-			ABT_cond_wait(cont->sc_rebuild_cond, cont->sc_mutex);
+		if (!daos_lru_is_last_user(&cont->sc_list))
+			ABT_cond_wait(cont->sc_fini_cond, cont->sc_mutex);
```

Comment:
Do we abort dtx resync & rebuild on container destroy or just wait for them done here?

I think if the dtx resync & rebuild just hold container for a short period, we could just retry the loop in function for few more times, if they hold container for a long period, we'd inform them to abort.

Looks not necessary to introduce this sc_fini_cond and hack the cont_child_put()?

Reply:
I think we abort resync, but for rebuild is a bit different because it is not based on container but based on pool. now the logic if we found container is destroying, we will skip that container scan/pulling before. but there that might take a bit longer if container has been held, we need wait existed operation finished..

Reply:
I think the rebuild code needs be changed to abort current iterating jobs (for scan or pull)?